keras_cv/models/densenet.py

```diff
@@ -0,0 +1,304 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""DenseNet models for KerasCV.
+
+Reference:
+  - [Densely Connected Convolutional Networks](https://arxiv.org/abs/1608.06993)
+  - [Based on the Original keras.applications DenseNet](https://github.com/keras-team/keras/blob/master/keras/applications/densenet.py)
+"""
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import backend
+from tensorflow.keras import layers
+
+BN_AXIS = 3
+
+
+def DenseBlock(x, blocks, name=None):
```

Comment:
This should be a wrapper, i.e. 

```python
def DenseBlock(blocks, name=None):
   if name is None:
       name = f"dense_block_{backend.get_uid("dense_block")}"
   def apply(x):
       for i in range(blocks):
            x = ConvBlock(x, 32, name=f"{name}_block_{i}")
       return x
   return apply
```

The way you can do `x = DenseBlock(3)(x)`